inference/core/workflows/core_steps/models/foundation/florence_2.py

```diff
@@ -0,0 +1,274 @@
+            "<REGION_TO_CATEGORY>",
+            "<REGION_TO_DESCRIPTION>",
+        ],
+        WorkflowParameterSelector(kind=[STRING_KIND]),
+    ] = Field(
+        description="The computer vision task to perform.",
+        default="<OPEN_VOCABULARY_DETECTION>",
+        examples=["<OPEN_VOCABULARY_DETECTION>"],
+    )
+    prompt: Union[WorkflowParameterSelector(kind=[LIST_OF_VALUES_KIND]), List[str]] = (
+        Field(
+            description="The accompanying prompt for the task (comma separated).",
+            examples=[["red apple", "blue soda can"], "$inputs.prompt"],
+        )
+    )
+
+    @classmethod
+    def accepts_batch_input(cls) -> bool:
+        return True
+
+    @classmethod
+    def describe_outputs(cls) -> List[OutputDefinition]:
+        return [
+            OutputDefinition(name="parent_id", kind=[BATCH_OF_PARENT_ID_KIND]),
+            OutputDefinition(name="root_parent_id", kind=[BATCH_OF_PARENT_ID_KIND]),
+            OutputDefinition(name="image", kind=[BATCH_OF_IMAGE_METADATA_KIND]),
+            OutputDefinition(name="raw_output", kind=[BATCH_OF_DICTIONARY_KIND]),
+            OutputDefinition(name="structured_output", kind=[BATCH_OF_DICTIONARY_KIND]),
+            OutputDefinition(name="predictions", kind=[BATCH_OF_DETECTION_KIND]),
+        ]
+
+    def get_actual_outputs(self) -> List[OutputDefinition]:
+        result = [
+            OutputDefinition(name="parent_id", kind=[BATCH_OF_PARENT_ID_KIND]),
+            OutputDefinition(name="root_parent_id", kind=[BATCH_OF_PARENT_ID_KIND]),
+            OutputDefinition(name="image", kind=[BATCH_OF_IMAGE_METADATA_KIND]),
+            OutputDefinition(name="raw_output", kind=[BATCH_OF_DICTIONARY_KIND]),
+            OutputDefinition(name="structured_output", kind=[BATCH_OF_DICTIONARY_KIND]),
+            OutputDefinition(name="predictions", kind=[BATCH_OF_DETECTION_KIND]),
+        ]
+        return result
+
+
+class Florence2ModelBlock(WorkflowBlock):
+
+    def __init__(
+        self,
+        model_manager: ModelManager,
+        api_key: Optional[str],
+        step_execution_mode: StepExecutionMode,
+    ):
+        self._model_manager = model_manager
+        self._api_key = api_key
+        self._step_execution_mode = step_execution_mode
+
+    @classmethod
+    def get_init_parameters(cls) -> List[str]:
+        return ["model_manager", "api_key", "step_execution_mode"]
+
+    @classmethod
+    def get_manifest(cls) -> Type[WorkflowBlockManifest]:
+        return BlockManifest
+
+    async def run(
+        self,
+        images: Batch[WorkflowImageData],
+        vision_task: str,
+        prompt: List[str],
+        model_id: str,
+    ) -> BlockResult:
+        if self._step_execution_mode is StepExecutionMode.LOCAL:
+            return await self.run_locally(
+                images=images, vision_task=vision_task, prompt=prompt, model_id=model_id
+            )
+        elif self._step_execution_mode is StepExecutionMode.REMOTE:
```

Comment:
I am not up to date with new models that are to be supported on the platform, in priv please provide me the info about models to be hosted on the platform and the way on how we want to host them - in particular - are we going to let this model to be run only locally / do we let people train the model etc.

Reply:
Like paligemma, florence 2 requires intensive compute to generate results in a reasonable amount of time so its only supported on local/dedicated deployments. we dont have a dedicated endpoint at this time. we currently support training loras outside of roboflow and uploading to the platform (and loading through this block)